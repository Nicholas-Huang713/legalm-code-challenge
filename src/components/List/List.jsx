import Button from "../Button/Button";
import "./List.scss";

export default function List({ list, handleBtnClick }) {
  return (
    <div className="list-container">
      <ul className="list-wrapper">
        {list.length > 0 ? (
          list.map((item) => (
            <li key={item.id} className="list-item">
              <div className="list-content">
                <div>
                  <strong>{item.name}</strong>
                </div>
                <div>Exp: {item.exp}yrs</div>
                <div>
                  <Button
                    btnText="X"
                    handleClick={handleBtnClick}
                    value={item.id}
                  />
                </div>
              </div>
              <div className="list-btn-wrapper">
                <Button
                  btnText="Details"
                  handleClick={handleBtnClick}
                  value={item.dogId}
                />
                <Button
                  btnText="Edit"
                  handleClick={handleBtnClick}
                  value={item.id}
                />
              </div>
            </li>
          ))
        ) : (
          <div>Adopt a Dog</div>
        )}
      </ul>
    </div>
  );
}
