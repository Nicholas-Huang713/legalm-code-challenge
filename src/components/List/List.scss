.list-container {
  max-width: 600px;
  margin: 0 auto;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 0 4px 8px rgba(251, 250, 250, 0.1);
}

.list-wrapper {
  padding: 0;
  border-radius: 4px;
  margin-top: 10px;
  height: 100%;
  width: 100%;
}

.list-item {
  border-bottom: 1px solid gray;
  padding: 5px;
}

.list-content {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  align-items: center;
  padding-bottom: 2px;
}

.list-btn-wrapper {
  display: flex;
  justify-content: space-evenly;
  flex-wrap: wrap;
}
