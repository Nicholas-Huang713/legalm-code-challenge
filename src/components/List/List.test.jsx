import { render, screen } from "@testing-library/react";
import List from "./List";

const mockOwners = [
  { id: 1, name: "Alice", exp: 10 },
  { id: 2, name: "Bob", exp: 5 },
  { id: 3, name: "Jerry", exp: 3 },
];

const onClickMock = vi.fn();
const renderPopulatedList = () => {
  return render(<List list={mockOwners} handleBtnClick={onClickMock} />);
};
const renderEmptyList = () => {
  return render(<List list={[]} handleBtnClick={onClickMock} />);
};

describe("List component", () => {
  it("renders list component - populated list", () => {
    const { container } = renderPopulatedList();
    const listItems = screen.getAllByRole("listitem");
    expect(listItems).toHaveLength(3);
    const detailsBtnList = screen.getAllByText("Details");
    expect(detailsBtnList).toHaveLength(3);
    const editBtnList = screen.getAllByText("Edit");
    expect(editBtnList).toHaveLength(3);
    expect(screen.getByText("Alice")).toBeInTheDocument();
    expect(screen.getByText("Exp: 10yrs")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders empty list message - empty list", () => {
    renderEmptyList();
    expect(screen.getByText("Adopt a Dog")).toBeInTheDocument();
  });
});
