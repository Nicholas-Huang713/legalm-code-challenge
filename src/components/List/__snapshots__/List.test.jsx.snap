// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`List component > renders list component - populated list 1`] = `
<div>
  <div
    class="list-container"
  >
    <ul
      class="list-wrapper"
    >
      <li
        class="list-item"
      >
        <div
          class="list-content"
        >
          <div>
            <strong>
              Alice
            </strong>
          </div>
          <div>
            Exp: 
            10
            yrs
          </div>
          <div>
            <button
              value="1"
            >
              X
            </button>
          </div>
        </div>
        <div
          class="list-btn-wrapper"
        >
          <button>
            Details
          </button>
          <button>
            Edit
          </button>
        </div>
      </li>
      <li
        class="list-item"
      >
        <div
          class="list-content"
        >
          <div>
            <strong>
              Bob
            </strong>
          </div>
          <div>
            Exp: 
            5
            yrs
          </div>
          <div>
            <button
              value="2"
            >
              X
            </button>
          </div>
        </div>
        <div
          class="list-btn-wrapper"
        >
          <button>
            Details
          </button>
          <button>
            Edit
          </button>
        </div>
      </li>
      <li
        class="list-item"
      >
        <div
          class="list-content"
        >
          <div>
            <strong>
              Jerry
            </strong>
          </div>
          <div>
            Exp: 
            3
            yrs
          </div>
          <div>
            <button
              value="3"
            >
              X
            </button>
          </div>
        </div>
        <div
          class="list-btn-wrapper"
        >
          <button>
            Details
          </button>
          <button>
            Edit
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
